### PR TITLE
Add Tests for ChipWithIcon Component issue #39

### LINKED ISSUE #39: Add component tests
## Description
Currently, only `Header` and `Footer` have component tests. However, these tests are naive. We need to update them and create new tests for all other components.

## TODO
Add tests for these component: https://github.com/nightwatchjs/html-reporter/tree/main/src/components

## What do sample tests look like?

Let's see the Header component
localhost_5173_

Here, we can assert the presence and visibility of the Nightwatch.js logo. We can also verify that the "Test Report", date, and time are being rendered.

## Reference
https://nightwatchjs.org/guide/component-testing/write-jsx-react-tests.html

### PR DESCRIPTION
**Pull request by aliamerj**
This pull request introduces comprehensive tests for the `ChipWithIcon` component to ensure it renders correctly under various conditions. The new tests verify that the component behaves as expected when different icons and text transformation properties are applied.

Key highlights include:
- Testing the rendering of multiple Material-UI icons within the component.
- Validating the text transformation logic based on the `transformText` prop.
- Ensuring the component maintains its structure and functionality with varied props.

Resolves #39

### CODE REVIEW

File: nightwatch/components/chipWithIcon.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ChipWithIconComponent.test = async (browser, { component }) => {
+  browser.expect(component).to.be.visible;
+  browser.expect(component).text.to.equal('Sample Chip');
+  browser.expect.element('[data-testid="chip-icon"]').to.be.present;
+};
+
+ChipWithIconComponentWithTransformText.test = async (browser, { component }) => {
+  browser.expect(component).to.be.visible;
+  browser.expect.element('[data-testid="chip-icon"]').to.be.present;
+  browser.expect(component).text.to.equal('sample chip');
```
Comment (vaibhavsingh97) [inline nightwatch/components/chipWithIcon.spec.tsx:37]: `transformText` converts `sample chip` to `Sample Chip`

Comment (aliamerj) [reply nightwatch/components/chipWithIcon.spec.tsx:37]: do i need to add other test with "sample chip" and expect "Sample Chip" ??

Comment (aliamerj) [reply nightwatch/components/chipWithIcon.spec.tsx:37]: or just change the values and keep it two tests ??

Comment (aliamerj) [reply nightwatch/components/chipWithIcon.spec.tsx:37]: when enable the transformText it did not convert sample chip to Sample chip
as you can see
image

**Pull request closed without merging**